epan/dissectors/packet-ippusb.c

```diff
@@ -21,70 +21,421 @@
+
+                    fragment_head * head = fragment_add_check(&ippusb_reassembly_table, tvb, offset, pinfo, new_msp->first_frame,
+                                                        GUINT_TO_POINTER(new_msp->first_frame), previous_msp->running_size, reported_length, FALSE);
+                    tvbuff_t *processedTvb = process_reassembled_data(tvb, offset, pinfo, "Reassembled IPPUSB", head, &ippusb_frag_items, NULL, tree);
+
+                    pinfo->can_desegment = 0;
+
+                    ret = dissector_try_uint_new(ippusb_dissector_table, HTTP, processedTvb, pinfo, tree, TRUE, data);
+                    col_append_fstr(pinfo->cinfo, COL_INFO, "  Reassembled Data");
+                    ippusb_last_pdu = -1;
+
+                }
+            }
+        }
+
+        pinfo->fragmented = save_fragmented;
+    }
+    else {
+
+        //not the first time this is seen
+
+        gboolean save_fragmented = pinfo->fragmented;
+        pinfo->fragmented = TRUE;
+        current_msp = (struct ippusb_multisegment_pdu *)wmem_tree_lookup32_le(ippusbd->multisegment_pdus, pinfo->num);
+
+        if (current_msp && current_msp->nxtpdu == 0 && !current_msp->finished) {
+             ti = proto_tree_add_item(tree, proto_ippusb, tvb, offset, -1, 0);
+            ippusb_tree = proto_item_add_subtree(ti, ett_ippusb);
+
+            fragment_head * head = fragment_get_reassembled_id(&ippusb_reassembly_table, pinfo, current_msp->first_frame);
+
+            if (head) {
+            tvbuff_t *processedTvb = process_reassembled_data(tvb, offset, pinfo, "Reassembled IPPUSB", head, &ippusb_frag_items, NULL, tree);
+
+            pinfo->can_desegment = 0;
+
+            ret = dissector_try_uint_new(ippusb_dissector_table, HTTP, processedTvb, pinfo, tree, TRUE, data);
+            col_append_fstr(pinfo->cinfo, COL_INFO, "  Reassembled Data with Error");
+            }
+
+        }
+        else if (current_msp && last == 0x03) {
+
+            ti = proto_tree_add_item(tree, proto_ippusb, tvb, offset, -1, 0);
+            ippusb_tree = proto_item_add_subtree(ti, ett_ippusb);
+
+            fragment_head * head = fragment_get_reassembled_id(&ippusb_reassembly_table, pinfo, current_msp->first_frame);
+
+            tvbuff_t *processedTvb = process_reassembled_data(tvb, offset, pinfo, "Reassembled IPPUSB", head, &ippusb_frag_items, NULL, tree);
+
+            pinfo->can_desegment = 0;
+
+            ret = dissector_try_uint_new(ippusb_dissector_table, HTTP, processedTvb, pinfo, tree, TRUE, data);
+            col_append_fstr(pinfo->cinfo, COL_INFO, "  Reassembled Data");
+
+        }
+
+        pinfo->fragmented = save_fragmented;
+    }
+
+    if (ret) {
+        return tvb_captured_length(tvb);
+    }
+    else {
+        return 0;
     }
-    return tvb_captured_length(tvb);
+}
+
+static int
+is_http_header(int first_linelen, const guchar *first_line) {
+    if ((first_linelen >= 5 && strncmp(first_line, "HTTP/", 5) == 0) ||
```

Comment:
>HTTP/ [](start = 52, length = 5)

Maybe we can get away with just this one and POST (all the IPP traffic requests are POSTs)?